staging/src/k8s.io/apiserver/pkg/server/genericapiserver.go

```diff
@@ -264,6 +277,8 @@ func (s preparedGenericAPIServer) Run(stopCh <-chan struct{}) error {
 		s.GenericAPIServer.AuditBackend.Shutdown()
 	}
 
+	s.RunPreStopHooks()
```

Comment:
The term "pre stop" is confusing. We stop the apiserver `Run` method and **after** that call the pre-stop hooks. So technically we have post-stop-hooks, or pre-shutdown-hooks.

Reply:
I originally had it called pre-shutdown hooks.  I think that is a pretty clear name for them and will change to reflect